### PR TITLE
DDF-05058 Refactors navigation-right to be extensible via extension points

### PR DESCRIPTION
**Pull request by vinamartin**
#### What does this PR do?
Currently navigation-right is extensible via plugins (216036c), it would be nice to move to extension points without breaking code bases that use plugins
#### Who is reviewing it? 
@Bdthomson @samuelechu 
#### Ask 2 committers to review/merge the PR and tag them here.

@adimka
@djblue 
#### How should this be tested?
Add a plugin to override the navigation right functionality. 
Add an override for navigationRight in `index.tsx`'s  `Entry()`. 
Example: 
```
{
  navigationRight: [() => <div>Some Text That Replaces Right Nav</div>],
}
```

#### Any background context you want to provide?
This is already extensible via plugins (216036c)

#### What are the relevant tickets?
Fixes: #5058 

#### Screenshots
Screen Shot 2019-07-02 at 11 31 21
_Navigation right_

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by vinamartin** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigation-right/navigation-right.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  Settings,
+  Notifications,
+  User,
+} from '../../react-component/presentation/navigation-right/buttons'
+
+import { SFC } from '../../react-component/hoc/utils'
+
+const DefaultItems = [Help, Settings, Notifications, User]
+
+export default plugin(DefaultItems) as SFC<any>[]
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigation-right/navigation-right.tsx]: nice

**Comment by vinamartin** [comment]
build now

**Comment by vinamartin** [comment]
build now

**Pull request merged**